cirq/ops/common_gates.py

```diff
@@ -982,3 +982,74 @@ def Ry(rads: float) -> RotYGate:
 def Rz(rads: float) -> RotZGate:
     """Returns a gate with the matrix e^{-i Z rads / 2}."""
     return RotZGate(rads=rads, global_shift_in_half_turns=-0.5)
+
+
+class XXPowGate(eigen_gate.EigenGate,
+             gate_features.TwoQubitGate,
+             gate_features.InterchangeableQubitsGate):
+    """
+    The gate implements the following unitary:
+    X⊗X = [0 0 0 1]
+          [0 0 1 0]
+          [0 1 0 0]
+          [1 0 0 0]
+
+    """
+
+    def __init__(self, *,  # Forces keyword args.
+                 exponent: Optional[Union[value.Symbol, float]] = None) -> None:
+        """
+        Initializes the gate.
+
+        The exponent may be specified.
+        If no argument is given, the default value of
+        exponent = 1 is used.
+
+        Args:
+            exponent: The power that the gate operation will be raised to.
+        """
+
+        super().__init__(exponent=value.chosen_angle_to_half_turns(half_turns=exponent))
+
+    @property
+    def exponent(self) -> Union[value.Symbol, float]:
+        return self._exponent
+
+    def _eigen_components(self):
+        return [
+            (1., np.array([[0.5, 0, 0, -0.5],
+                            [0, 0.5, -0.5, 0],
+                            [0, -0.5, 0.5, 0],
+                            [-0.5, 0, 0, 0.5]])),
+            (0., np.array([[0.5, 0, 0, 0.5],
+                            [0, 0.5, 0.5, 0],
+                            [0, 0.5, 0.5, 0],
+                            [0.5, 0, 0, 0.5]]))
+        ]
+
+    def _canonical_exponent_period(self) -> Optional[float]:
+        return 2
+
+    def _with_exponent(self,
+                       exponent: Union[value.Symbol, float]) -> 'XXPowGate':
+        return XXPowGate(exponent=exponent)
+
+    def _circuit_diagram_info_(self, args: protocols.CircuitDiagramInfoArgs
+                               ) -> protocols.CircuitDiagramInfo:
+        return protocols.CircuitDiagramInfo(
```

Comment:
Add

```python
    if self._global_shift == -0.5:
            return _rads_func_symbol(
                'MS',
                args,
                self._diagram_exponent(args, ignore_global_phase=False))
```

to the start of this method.

Check out https://github.com/quantumlib/Cirq/pull/1101/files#diff-08a88625437f21ce336680722c6942d5R104 where this method is defined. Note that that PR is not yet merged.

Reply:
Sorry for the bad code before:

```
 if self._global_shift == -0.5:
    symbol = _rads_func_symbol(
        'MS',
        args,
        self._diagram_exponent(args, ignore_global_phase=False))
    return symbol, '#2'  
```